### ISSUE TITLE
Dynamic zone not inserting items correctly when using "Add component below/above" button

### ISSUE THREAD

**Issue by othecos** [strapi/strapi#22399]
### Required System information

- Node.js version: 22.12.0
- NPM version: 10.9.0
- Strapi version: 5.5.1
- Database: MySql
- Operating system: MacOs
- Is your project Javascript or Typescript: Typescript

### Describe the bug

When adding new items to a dynamic zone list, clicking 'Add Component Below' replaces the item below the selected element instead of inserting a new item underneath it as expected. Similarly, clicking 'Add Component Above' replaces the current item with the new one, rather than adding it above the selected element.

### Steps to reproduce the behavior

1. Create components to add to the dynamicZone list
2. Create a collection type with a dynamicZone attribute
3. Add those components in the dynamicZone.
4. Create a new item on the new collection.
5. Upon creation, add components to the dynamicZone
6. Try to add components above or below on that dynamicZone list using the buttons provided by the admin panel (3 dots)
7. (To reproduce the issue, the component you are referencing to insert a new item above or below should already have items below and above)

### Expected behavior

- New components added via "Add component above" should be inserted **above** the referenced component where the action was triggered.
- New components added via "Add component below" should be inserted **below** the referenced component where the action was triggered.

### Screenshots

https://github.com/user-attachments/assets/ff49eec5-54c6-4b88-89b7-69ba00cb541e

### Code snippets

If applicable, add code samples to help explain your problem.

**Comment by othecos** [comment]
This is very high priority and it's preventing us from migrating from v4 to v5, since dynamic zones are a crucial part of our CMS usage.

**Comment by jannescb** [comment]
+1

**Comment by jannescb** [comment]
How can this only be medium? This DELETES content

**Comment by Convly** [comment]
Hey everyone, thanks for reporting this. I've been able to reproduce the issue.

We've bumped the issue severity as it breaks the product usage, and we've started looking actively into it.

> How can this only be medium? This DELETES content

This won't _actually_ delete anything until you save your changes. If you reload the page, it should restore the correct components. But nonetheless, we agree this is quite an important issue as it might lead to content loss in case this is not spotted before saving the data.

We will report on this when we have more information or a fix to share with you all.

**Comment by ismoki** [comment]
This was reported earlier in issue: https://github.com/strapi/strapi/issues/22154

**Comment by derrickmehaffy** [comment]
Raising Sev to critical due to the data loss, implementing our internal escalation policy.

**Comment by Convly** [comment]
> Any expectations when this will be fixed  ?

The code causing the issue has been identified, and the content experience team is expected to look into it starting this week. Then it'll depend on the complexity to issue a fix.

Sorry for any inconvenience you're encountering in the meantime

**Comment by derrickmehaffy** [comment]
The latest information from the engineering manager is they will hopefully be able to take a look this week. Today they were down to only 1 engineer who is working on something else but we should have a few more engineers coming back this week to take a look.

**Comment by rehan509** [comment]
hi @derrickmehaffy  Any update on this issue ,has this been fixed ?

**Comment by derrickmehaffy** [comment]
> hi @derrickmehaffy  Any update on this issue ,has this been fixed ?
> 

Yup see the linked PRs just above your comment. It was already fixed and released
